### PR TITLE
Automated Cypress testing

### LINKED ISSUE #226: Automated testing
**tools**
- Cypress
- Cucumber

**test steps**
- [ ] home page lists multiple users
- [ ] click on first/random user and check the data display is correct according to their json file

### PR DESCRIPTION
**Pull request by WendallHersh**
resolve #226

Added Cypress and Typescript

Creates Tests for HomePage and User Page (eddiejaoude) <--- seems this one will always be there...

This is just the initial setup for Cypress and can be worked upon to create additional tests. 
Tests do not currently run automatically but can be run from the main directory with the following command 

` ./node_modules/.bin/cypress open
`

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/305"></a>

### CODE REVIEW

File: cypress/integration/users/users.spec.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/// <reference types="cypress" />
+import userData from '../../../public/data/eddiejaoude.json'
+
+const baseUrl = 'http://linkfree.eddiehub.org';
+
+context('Users', () => {
+    beforeEach(() => {
+      cy.visit(`${baseUrl}/eddiejaoude`,)
```
Comment (eddiejaoude) [inline cypress/integration/users/users.spec.ts:8]: Could this be done as a `click` from the homepage to check the navigating from the homepage to the user page works?

File: package.json [hunk 2]
```diff
@@ -47,13 +47,15 @@
...
     "eslint": "^7.32.0",
     "eslint-config-standard": "^16.0.3",
     "eslint-plugin-import": "^2.24.2",
     "eslint-plugin-json": "^3.1.0",
     "eslint-plugin-node": "^11.1.0",
     "eslint-plugin-promise": "^5.1.0",
     "eslint-plugin-react": "^7.25.1",
-    "prettier": "2.4.0"
+    "prettier": "2.4.0",
+    "typescript": "^4.4.3"
```
Comment (eddiejaoude) [inline package.json:59]: I really like CucumberJS in addition to Cypress. Is this something you can add? If not it can be added in another PR

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
File line endings are marked with red mark by GitHub, they need a blank ending line

**Comment by eddiejaoude** [comment]
Thank you for the PR, looks great 👍 left some small comments

File: cypress/integration/homepage/homepage.spec.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+context('Homepage', () => {
+    beforeEach(() => {
+      cy.visit(baseUrl);
+    })
+    
+    it('lists multiple users', () => {
+        cy.get('main a').should('have.length.at.least',2);
+    })
+  })
+
```
Comment (Panquesito7) [inline cypress/integration/homepage/homepage.spec.ts:14]: Missing an endline.
```suggestion

```

File: cypress/integration/users/users.spec.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    })
+    
+    it('page shows content from user provided file', () => {
+        cy.url().should('include', '/eddiejaoude')
+        cy.get('h1').should('contain.text',userData.name)
+        cy.get('main > section > div').should('contain.text',userData.bio)
+      })
+  })
+
```
Comment (Panquesito7) [inline cypress/integration/users/users.spec.ts:17]: Same here.

File: cypress/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+    "compilerOptions": {
+      "target": "es5",
+      "lib": ["es5", "dom"],
+      "types": ["cypress"],
+      "resolveJsonModule": true,
+      "esModuleInterop": true,
+    },
+    "include": ["**/*.ts"]
+  }
```
Comment (Panquesito7) [inline cypress/tsconfig.json:10]: Same here.
```suggestion
}

```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (WendallHersh) [reply package.json:59]: Cucumber is not something I am comfortable with quite yet and would take a few days IU think to implement by me. Maybe something I could do for Hacktober or something.

Comment (eddiejaoude) [inline package.json:59]: Is typescript needed? I don't see it being used?

**Comment by eddiejaoude** [comment]
Thank you for making those changes, sorry I had 1 last question

File: package.json (hunk 2 above)
Comment (WendallHersh) [reply package.json:59]: It looks like it is needed for the cypress UI to run. I tried to remove it but the Cypress UI was not functional.

**Comment by eddiejaoude** [comment]
Nooooo I changed the branch not closed 😢

**Comment by eddiejaoude** [comment]
Why can't I re-open it 😱

**Comment by eddiejaoude** [comment]
Sorry @WendallHersh I messed this up somehow, I think I can create another PR from your fork so your commits are still there

**Comment by eddiejaoude** [comment]
I managed to recreate a PR from your fork https://github.com/EddieHubCommunity/LinkFree/pull/380

**Pull request closed without merging**